### PR TITLE
Adding Gradle Support to jpf-symbc GSoC'22 Project

### PR DESCRIPTION
**Pull request by gaurangkudale**
Hey there,

Here I have done several changes as follows -

1. Added Gradle support to jpf-symbc.
2. Added jpf-core as a Git-Submodule of SPF and jpf-symbc as a normal project.
3. I have restructured the project as 
SPF | (Root-project)
                      -----| jpf-core (Git-Submodule & Gradle subproject)
                      -----| jpf-symbc (Gradle subproject) 
4. Upgraded the Gradle version to 6.9
5. Added / Pulled dependencies from the maven central repository to reduce the Jars in lib dir.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -0,0 +1,15 @@
+apply plugin: "java-library"
+apply plugin: "java"
+
+
+task copyAllJars(type: Copy) {
```
Comment (yannicnoller) [inline build.gradle:5]: The copy task should be removed.

File: build.gradle [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+    group = "Copy Jars"
+    description = "Copies the jpf-core and jpf-symbc JARS to SPF/build/allJars."
+
+    from subprojects.collect { it.tasks.withType(Jar) }
+    into "$buildDir/allJars"
+
+}
+
+defaultTasks "copyAllJars"
```
Comment (yannicnoller) [inline build.gradle:15]: Can we add a command or script that handles all building and testing?

File: jpf-symbc/build.gradle [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+                                 "${result.successfulTestCount} passed",
+                                 "${result.failedTestCount} failed",
+                                 "${result.skippedTestCount} skipped"]
+
+            println "Summary: " + summaryFields.join(", ")
+        }
+    }
+}
+
+publishing {
```
Comment (yannicnoller) [inline jpf-symbc/build.gradle:172]: Add a comment about where jars will be published. Generally, the file could be documented more.

File: jpf-symbc/build.gradle [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+
+    dependsOn compile
+    dependsOn copyResources
+
+    from sourceSets.annotations.java.outputDir
+}
+
+task buildJars {
+    group = "SPF Build"
+    description = "Generates all JPF symbc jar files."
```
Comment (yannicnoller) [inline jpf-symbc/build.gradle:123]: change to "SPF jar files" or "jpf-symbc jar files"

File: jpf-symbc/build.gradle [hunk 5]
```diff
@@ -0,0 +1,192 @@
+apply plugin: "java"
+apply plugin: "maven-publish"
+
+def getCurrentVersion() {
```
Comment (yannicnoller) [inline jpf-symbc/build.gradle:4]: remove hardcoded version and use something similar to JPF: "JPF-8.0-112-ge91ea1f-SNAPSHOT"

File: .github/workflows/gradle-publish.yml [hunk 6]
```diff
@@ -0,0 +1,33 @@
+name: Gradle Package
```
Comment (yannicnoller) [inline .github/workflows/gradle-publish.yml:1]: rename the file, e.g., to "main.yml"

File: .github/workflows/gradle-publish.yml [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
```
Comment (yannicnoller) [inline .github/workflows/gradle-publish.yml:2]: improve the format of the file, e.g., by adding blank lines, documentation, a,d indentation.

**Review by yannicnoller — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 1 above)
Comment (gaurangkudale) [reply build.gradle:5]: This comment is for the user to copyAllJars from `jpf-core/build` `jpf-symbc/build` to `SPF/build`
For example, if the user wants all Jars to present in the `SPF/build` dir then the user can fire this task manually (Note: this task is not copying the jpf-core jars to jpf-symbc/lib)
If we do not want this task then I can remove it.

File: jpf-symbc/build.gradle (hunk 5 above)
Comment (gaurangkudale) [reply jpf-symbc/build.gradle:4]: Hi @yannicnoller
I have gone through the build.gradle file of the jpf-core but have'nt figureout how they are using git-hash and java version to deploy it locally and I'll contact Mr.Cyrill regarding this.

File: jpf-symbc/build.gradle (hunk 4 above)
Comment (gaurangkudale) [reply jpf-symbc/build.gradle]: Done

File: jpf-symbc/build.gradle (hunk 3 above)
Comment (gaurangkudale) [reply jpf-symbc/build.gradle:182]: Done

File: build.gradle (hunk 1 above)
Comment (gaurangkudale) [reply build.gradle]: I have removed this task

**Comment by gaurangkudale** [comment]
- [ ]  README is empty
- [ ]  at least add the instructions for the build
- [ ]  build files
    - [ ]  we still have the copy task?
    [https://github.com/gaurangkudale/SPF/blob/ba7369bce2b6324a847112b58b864a2ceceb55c6/build.gradle#L5](https://github.com/gaurangkudale/SPF/blob/ba7369bce2b6324a847112b58b864a2ceceb55c6/build.gradle#L5)

- [ ] can we have one gradle command for everything, e.g., have a task in the SPF/build.gradle that triggers the builds for the core and symbc… or have a simply make file, so that someone only has to execute make?
- [ ]  the publishing at the end of SPF/jpf-smbc/build.gradle is that needed?
[https://github.com/gaurangkudale/SPF/blob/ba7369bce2b6324a847112b58b864a2ceceb55c6/jpf-symbc/build.gradle#L172-L190](https://github.com/gaurangkudale/SPF/blob/ba7369bce2b6324a847112b58b864a2ceceb55c6/jpf-symbc/build.gradle#L172-L190)
    - [ ]  add a comment that is local maven repo
        - [ ]  will be triggered by specific commands

- [ ] change to “SPF jar files@” → [https://github.com/gaurangkudale/SPF/blob/ba7369bce2b6324a847112b58b864a2ceceb55c6/jpf-symbc/build.gradle#L123](https://github.com/gaurangkudale/SPF/blob/ba7369bce2b6324a847112b58b864a2ceceb55c6/jpf-symbc/build.gradle#L123)
    - [ ]  `jpf-symbc` or `SPF`

- [ ] Can we run build and tests and separate steps (core and spf)?
https://github.com/nus-se/its-baseline/blob/master/.github/workflows/main.yml

- [ ] can we separate spf build and test, improve format of the file, name of .yml file: main

**Pull request closed without merging**